crates/rattler_networking/src/authentication_storage/storage.rs

```diff
@@ -0,0 +1,183 @@
+//! Storage and access of authentication information
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    str::FromStr,
+    sync::{Arc, Mutex},
+};
+
+use keyring::Entry;
+use reqwest::{IntoUrl, Url};
+
+use super::{authentication::Authentication, fallback_storage};
+
+/// A struct that implements storage and access of authentication
+/// information
+#[derive(Clone)]
+pub struct AuthenticationStorage {
+    /// The store_key needs to be unique per program as it is stored
+    /// in a global dictionary in the operating system
+    pub store_key: String,
+
+    /// Fallback JSON location
+    pub fallback_json_location: PathBuf,
```

Comment:
Why not use the `FallbackStorage` object here, instead of recreating it where you need it?